### PR TITLE
Do not allow dob entry for future dates

### PR DESCRIPTION
**Pull request by Phemus**
Fix html validatation to allow dates newer than 2014. also added validation to prevent future dates

### CODE REVIEW

File: application/templates/enterdob.html [hunk 1]
```diff
@@ -37,7 +37,7 @@ <h1 class="heading-small">What is your date of birth?</h1>
 
           <div class="form-group form-group-year">
             <label for="dob-year">Year</label>
-            <input class="form-control" id="dob-year" name="dob-year" type="number" pattern="[0-9]*" min="0" max="2014">
```
Comment (danielmarker) [inline application/templates/enterdob.html:40]: Current Year -2 to meet the previous rule?

File: application/deed/searchdeed/views.py [hunk 2]
```diff
@@ -32,6 +37,7 @@ def validate_dob(form):
 @searchdeed.route('/enter-dob', methods=['POST'])
 def enter_dob():
     form = request.form
+    form.current_year = str(datetime.datetime.now().year)
```
Comment (danielmarker) [inline application/deed/searchdeed/views.py:39]: Is it correct to be instantiated as an object of form?

**Comment by tsangsimon** [comment]
:+1:

File: application/deed/searchdeed/views.py [hunk 3]
```diff
@@ -16,11 +16,16 @@ def search_deed_main():
...
         year = int(form["dob-year"])
 
-        datetime.datetime(year, month, day)
+        dob_date = datetime.datetime(year, month, day)
+
+        if dob_date >= present:
+            raise Exception("Date cannot be in the future")
 
     except:
         print(sys.exc_info()[0])
```
Comment (tsangsimon) [inline application/deed/searchdeed/views.py]: use the logger please

**Comment by tsangsimon** [comment]
thanks for adding the unit tests :+1:

**Pull request merged**